### ISSUE TITLE
Installing loadcaffe by specifying protobuf path?

### ISSUE THREAD

**Issue by tanelp** [szagoruyko/loadcaffe#60]
I have two versions of protobuf installed (version 2.3.0 has been pre-installed in the server, and I have installed version 2.6.1 locally), but cmake uses the older version which is probably why I cannot install loadcaffe.

``` 
$ luarocks install loadcaffe
Installing https://raw.githubusercontent.com/torch/rocks/master/loadcaffe-1.0-0.rockspec...
Using https://raw.githubusercontent.com/torch/rocks/master/loadcaffe-1.0-0.rockspec... switching to 'build' mode
Initialized empty Git repository in /tmp/luarocks_loadcaffe-1.0-0-2611/loadcaffe/.git/
remote: Counting objects: 23, done.
remote: Compressing objects: 100% (21/21), done.
remote: Total 23 (delta 4), reused 9 (delta 1), pack-reused 0
Receiving objects: 100% (23/23), 24.83 KiB, done.
Resolving deltas: 100% (4/4), done.
cmake -E make_directory build;
cd build;
cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH="/home/hpc_tanel/torch/install/bin/.." -DCMAKE_INSTALL_PREFIX="/home/hpc_tanel/torch/install/lib/luarocks/rocks/loadcaffe/1.0-0";
make

-- The C compiler identification is GNU 4.8.2
-- The CXX compiler identification is GNU 4.8.2
-- Check for working C compiler: /opt/gcc-4.8.2/bin/gcc
-- Check for working C compiler: /opt/gcc-4.8.2/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /opt/gcc-4.8.2/bin/g++
-- Check for working CXX compiler: /opt/gcc-4.8.2/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Found Torch7 in /home/hpc_tanel/torch/install
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found PROTOBUF: /usr/lib64/libprotobuf.so
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/luarocks_loadcaffe-1.0-0-2611/loadcaffe/build
[ 33%] Running C++ protocol buffer compiler on caffe.proto
Scanning dependencies of target loadcaffe
[ 66%] Building CXX object CMakeFiles/loadcaffe.dir/loadcaffe.cpp.o
In file included from /tmp/luarocks_loadcaffe-1.0-0-2611/loadcaffe/loadcaffe.cpp:21:0:
/tmp/luarocks_loadcaffe-1.0-0-2611/loadcaffe/build/caffe.pb.h:12:2: error: #error This file was generated by a newer version of protoc which is
 #error This file was generated by a newer version of protoc which is
  ^
/tmp/luarocks_loadcaffe-1.0-0-2611/loadcaffe/build/caffe.pb.h:13:2: error: #error incompatible with your Protocol Buffer headers. Please update
 #error incompatible with your Protocol Buffer headers.  Please update
  ^
/tmp/luarocks_loadcaffe-1.0-0-2611/loadcaffe/build/caffe.pb.h:14:2: error: #error your headers.
 #error your headers.
  ^
In file included from /tmp/luarocks_loadcaffe-1.0-0-2611/loadcaffe/loadcaffe.cpp:21:0:
/tmp/luarocks_loadcaffe-1.0-0-2611/loadcaffe/build/caffe.pb.h:26:55: fatal error: google/protobuf/generated_enum_reflection.h: No such file or directory
 #include <google/protobuf/generated_enum_reflection.h>
                                                       ^
compilation terminated.
make[2]: *** [CMakeFiles/loadcaffe.dir/loadcaffe.cpp.o] Error 1
make[1]: *** [CMakeFiles/loadcaffe.dir/all] Error 2
make: *** [all] Error 2

Error: Build error: Failed building.
```

Additional information:
```
$ protoc --version
libprotoc 2.6.1
```

```
$ /usr/bin/protoc --version
libprotoc 2.3.0
```

**Comment by szagoruyko** [comment]
you should tell cmake to use newer protobuf. this should help https://cmake.org/cmake/help/v3.0/module/FindProtobuf.html

**Comment by tanelp** [comment]
Thanks! I set PROTOBUF_LIBRARY and PROTOBUF_INCLUDE_DIR variables in the CMakeLists.txt file, and now it works.

```
SET(PROTOBUF_LIBRARY "/home/hpc_tanel/lib/libprotobuf.so")
SET(PROTOBUF_INCLUDE_DIR "/home/hpc_tanel/include")
```

**Comment by sidgan** [comment]
@szagoruyko @tanelp I am getting a similar error: 

`luarocks install loadcaffe-1.0-0.rockspec
Using loadcaffe-1.0-0.rockspec... switching to 'build' mode
Cloning into 'loadcaffe'...
remote: Counting objects: 18, done.
remote: Compressing objects: 100% (17/17), done.
remote: Total 18 (delta 0), reused 5 (delta 0), pack-reused 0
Unpacking objects: 100% (18/18), done.
cmake -E make_directory build;
cd build;
cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH="/home/sganju1/torch/install/bin/.." -DCMAKE_INSTALL_PREFIX="/home/sganju1/torch/install/lib/luarocks/rocks/loadcaffe/1.0-0";
make

-- The C compiler identification is GNU 4.9.3
-- The CXX compiler identification is GNU 4.9.3
-- Check for working C compiler: /home/xinleic/tools/gcc/bin/gcc
-- Check for working C compiler: /home/xinleic/tools/gcc/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /home/xinleic/tools/gcc/bin/g++
-- Check for working CXX compiler: /home/xinleic/tools/gcc/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Torch7 in /home/sganju1/torch/install
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found PROTOBUF: /home/xinleic/cools/protobuf/lib/libprotobuf.so
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/luarocks_loadcaffe-1.0-0-9052/loadcaffe/build
[ 25%] Running C++ protocol buffer compiler on caffe.proto
Scanning dependencies of target loadcaffe
[ 50%] Building CXX object CMakeFiles/loadcaffe.dir/loadcaffe.cpp.o
In file included from /tmp/luarocks_loadcaffe-1.0-0-9052/loadcaffe/loadcaffe.cpp:21:0:
/tmp/luarocks_loadcaffe-1.0-0-9052/loadcaffe/build/caffe.pb.h:12:2: error: #error This file was generated by a newer version of protoc which is
 #error This file was generated by a newer version of protoc which is
  ^
/tmp/luarocks_loadcaffe-1.0-0-9052/loadcaffe/build/caffe.pb.h:13:2: error: #error incompatible with your Protocol Buffer headers. Please update
 #error incompatible with your Protocol Buffer headers.  Please update
  ^
/tmp/luarocks_loadcaffe-1.0-0-9052/loadcaffe/build/caffe.pb.h:14:2: error: #error your headers.
 #error your headers.
  ^
In file included from /tmp/luarocks_loadcaffe-1.0-0-9052/loadcaffe/loadcaffe.cpp:21:0:
/tmp/luarocks_loadcaffe-1.0-0-9052/loadcaffe/build/caffe.pb.h:26:55: fatal error: google/protobuf/generated_enum_reflection.h: No such file or directory
 #include <google/protobuf/generated_enum_reflection.h>
                                                       ^
compilation terminated.
make[2]: *** [CMakeFiles/loadcaffe.dir/loadcaffe.cpp.o] Error 1
make[1]: *** [CMakeFiles/loadcaffe.dir/all] Error 2
make: *** [all] Error 2

Error: Build error: Failed building.`

Additional Information: 

`protoc --version
libprotoc 2.6.1`

`/usr/bin/protoc --version
libprotoc 2.3.0`

I have set the protobuf library paths in the CMakeLists.txt file as: 

`SET(PROTOBUF_LIBRARY "/home/xiaolonw/protobuf/lib/libprotobuf.so")
SET(PROTOBUF_INCLUDE_DIR "/home/xiaolonw/protobuf")` 

Anything else you would suggest?
